tests/CppWinRTAuthoringTests.cpp

```diff
@@ -226,73 +228,57 @@ TEST_CASE("CppWinRTAuthoringTests::EventsAndCppWinRt", "[property]")
 #include <winrt/Windows.System.h>
 #include <winrt/Windows.UI.Xaml.Hosting.h>
 
-// This test seems to have issues on the CI machines, so enabling only when run locally
-TEST_CASE("CppWinRTAuthoringTests::NotifyPropertyChanged", "[property][LocalOnly]")
+TEST_CASE("CppWinRTAuthoringTests::NotifyPropertyChanged", "[property]")
 {
 #if defined(WIL_ENABLE_EXCEPTIONS)
-    auto uninit = wil::RoInitialize_failfast(RO_INIT_MULTITHREADED);
+    auto uninit = wil::RoInitialize_failfast(RO_INIT_SINGLETHREADED);
     // We need to initialize the XAML core in order to instantiate a PropertyChangedEventArgs.
-    // Do all the work on a separate DispatcherQueue thread so we can shut it down cleanly

```

Comment:
bad merge? looks like it might be missing raymond's change to drain the queue?

Reply:
This intentionally reverted that change as it's no longer necessary since we're no longer using `IMallocSpy`. This also means that this test no longer sporadically fails and can therefore be enabled to run on CI machines.

Reply:
IIUC the xaml manager Close is async so technically you still need to drain the queue (it might happen to not crash, but it's a bug to not drain the queue)

Reply:
Fair point. I guess I can keep the dispatcher queue & thread, but remove the wait

Reply:
Hmm. For some reason even doing it this way causes issues when running on the CI machines. Need to take a deeper look at it...

Reply:
Oh my. The DispatcherQueue seems to intentionally crash if its DispatcherQueueController "parent" gets destroyed (destructor runs) when shutdown starts, and looks like it will unintentionally crash if the controller gets destroyed when shutdown completes. My guess is that the CI machines are hitting the latter scenario as it's reporting an AV, not a fail fast exception. From what I can gather, this should be safe if we wait for `ShutdownQueueAsync` to complete before completing the test.

Reply:
Well, it wasn't very smart of me to look at the _current_ implementation when the CI machines are running Windows Server 2022. From what I can gather from the build number, this correlates to fe_release. I created a VM with the same build (albeit Desktop, not Server) and was able to repro this under TTD. Looks like `InputHost.dll` is getting unloaded before the thread pool tries to invoke a callback into that dll on thread termination. I'm unsure if this is something that has been fixed since then. I've never hit this locally, so maybe it has, however it's tricky to say since the code in question doesn't exist anymore (though it's likely it's just been moved elsewhere). That said, the operation being scheduled is a `Release` call, so hopefully if it _has_ been moved, wherever it was moved to is using something like WRL and has proper module ref count tracking.

In the end, it looks like Raymond's change to wait for thread termination is probably the most reasonable way to work around this bug on both CI machines and older builds. I've increased the timeout so that hopefully the potato CI machines don't hit as many false positives.

Reply:
ooof. well, xaml islands is fun like that. 
@jessecol does this ring any bells?

Reply:
Hi!  It doesn't ring any bells, I've found that usually closing all the WindowsXamlManagers/DeskopWindowXamlSources and draining message queues on the threads using Xaml is enough for a clean shutdown.  Sadly, I can't recall what objects live in InputHost.dll or speculate intelligently about why it's getting unloaded.

Reply:
The function that registers the TP Wait is `InputHost!WRL2ContextSession::RegisterThreadExitCallback`. The CI machines seem to be running Windows Server 2022 build 10.0.20348, which AFAICT is a build number that correlates to `fe_release`, so pretty old; predates Windows 11. Wouldn't surprise me if this has been fixed - either intentionally or unintentionally because that code no longer seems to exist - and this hasn't really been seen/reported.